pennylane/ops/qubit/parametric_ops.py

```diff
@@ -837,18 +877,24 @@ def simplify(self):
         Hadamard(wires=[0])
 
         """
-        p0, p1, p2 = np.mod(self.data, 2 * np.pi)
-
-        if np.allclose(p0, np.pi / 2) and np.allclose(np.mod(self.data[2], -2 * np.pi), -np.pi / 2):
-            return qml.RX(self.data[1], wires=self.wires)
-        if np.allclose(p0, 0) and np.allclose(p2, 0):
-            return qml.RY(self.data[1], wires=self.wires)
-        if np.allclose(p1, 0):
-            return qml.RZ(self.data[0] + self.data[2], wires=self.wires)
-        if np.allclose(p0, np.pi) and np.allclose(p1, np.pi / 2) and np.allclose(p2, 0):
+        p0, p1, p2 = [p % (4 * np.pi) for p in self.data]
+
```

Comment:
Like in the other scenarios, we should make sure the parameters are not trainable before we simplify them out of existence.

Reply:
Whoops, my bad. 😅